src/Symfony/Component/Serializer/Normalizer/UidNormalizer.php

```diff
@@ -72,7 +73,7 @@ public function denormalize($data, string $type, string $format = null, array $c
         try {
             return Ulid::class === $type ? Ulid::fromString($data) : Uuid::fromString($data);
         } catch (\InvalidArgumentException $exception) {
-            throw new NotNormalizableValueException(sprintf('The data is not a valid "%s" string representation.', $type));
+            throw NotNormalizableValueException::create(sprintf('The data is not a valid "%s" string representation.', $type), $data, [Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
```

Comment:
also, $type is leaking a SF class (but i like this message generally)

Reply:
IMHO, This is not something sensitive. It only leak the app is written in PHP but it give a real hint for the end user.
WDYT?

Reply:
generally i'd say we talk "a valid UUID string", similar like we talk about a "datetime string"

Reply:
You were right, I changed this code and added another catch (type error, If I pass `null`)